pkg/virt-controller/watch/snapshot/source.go

```diff
@@ -148,12 +159,78 @@ func (s *vmSnapshotSource) Unlock() (bool, error) {
 	return true, nil
 }
 
-func (s *vmSnapshotSource) Spec() snapshotv1.SourceSpec {
-	vmCpy := s.vm.DeepCopy()
-	vmCpy.Status = kubevirtv1.VirtualMachineStatus{}
+func (s *vmSnapshotSource) HaveHotplug() (bool, error) {
+	vmi, exists, err := s.controller.getVMI(s.vm)
+	if err != nil || !exists {
+		return false, err
+	}
+
+	vmRevision, err := s.GetVMRevision()
+	if err != nil {
+		return false, err
+	}
+
+	vmiPVCs := getPVCsFromVolumes(vmi.Spec.Volumes)
+	vmRevisionPVCs := getPVCsFromVolumes(vmRevision.Spec.Template.Spec.Volumes)
+
+	return !reflect.DeepEqual(vmiPVCs, vmRevisionPVCs), nil
+}
+
+func (s *vmSnapshotSource) GetVMRevision() (*kubevirtv1.VirtualMachine, error) {
+	online, err := s.Online()
+	if err != nil || !online {
+		return nil, err
+	}
+
+	vmi, exists, err := s.controller.getVMI(s.vm)
+	if err != nil || !exists {
+		return nil, err
+	}
+
+	crName := vmi.Spec.VirtualMachineRevisionName
+	cr, err := s.controller.Client.AppsV1().ControllerRevisions(s.vm.Namespace).Get(context.TODO(), crName, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	vmRevision := &kubevirtv1.VirtualMachine{}
+	err = json.Unmarshal(cr.Data.Raw, vmRevision)
```

Comment:
since we are in charge of both encoding and decoding I think it is safe to marshall/unmarshall directly but you are correct in that this is the "right" way to do it

Reply:
I changed the encoding to be as done in statefulset and daemonset.